tests/pausing_test.go

```diff
@@ -373,7 +373,7 @@ var _ = Describe("[rfe_id:3064][crit:medium][vendor:cnv-qe@redhat.com][level:com
 
 		grepSleepPid := func(expecter expect.Expecter) string {
 			res, err := console.ExpectBatchWithValidatedSend(expecter, []expect.Batcher{
-				&expect.BSnd{S: `pgrep -f "sleep 8"` + "\n"},
```

Comment:
I'm not sure how useful this test originally was, but I think that this change made it worse as it makes the following statement untrue.
```
			By("Waiting longer than the process normally runs")
			time.Sleep(10 * time.Second)
```
Adding more sleep time to solve a race is a known fallacy. You should never do it without explaining why it only sweeps the problem under the rug rather than resolve it.

I think that a better test for pause would be to measure the difference in uptime between guest and host, pause, and measure again. It is still raceful but quicker. To avoid the race you can trust libvirt and simply ask it if the domain is paused.

Reply:
> I'm not sure how useful this test originally was, but I think that this change made it worse as it makes the following statement untrue.
> 
> ```
> 			By("Waiting longer than the process normally runs")
> 			time.Sleep(10 * time.Second)
> ```
> 
Thank you @dankenigsberg  for this catch, I totally missed it. https://github.com/kubevirt/kubevirt/pull/4394 should quick-fix it.
> Adding more sleep time to solve a race is a known fallacy. You should never do it without explaining why it only sweeps the problem under the rug rather than resolve it.

I agree, but I think that in this case it is not so bad:
The point of this test, as I understand it,  is to check if a process is continued after we pause the vmi. I don't think that testing the pausing action itself is in the scope of this test case.
`sleep` is just been used here as a long process, it could be any other process that runs for a long time.
anyway, It is not causing any other component to wait while something else happening, which might lead to unexpected behavior. (as sometimes happen when using `Eventually` too widely, for example).

The point of this change is to make sure the process doesn't finish before we pause the vmi. so by making it longer we can be more positive about it won't happen. I don't think it sweeps the problem under the rug in that case.
make sense, or am I missing something?

> 
> I think that a better test for pause would be to measure the difference in uptime between guest and host, pause, and measure again. It is still raceful but quicker. To avoid the race you can trust libvirt and simply ask it if the domain is paused.

Sounds good!. I can try and add a test  case to measure that, but I don't think it belongs in this `It` scope.
As for asking libvirt, I think this is done by [retrieving the condition](https://github.com/kubevirt/kubevirt/blob/master/tests/pausing_test.go#L422), which is eventually is [updating from the domain](https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/vm.go#L822). 

What do you think?

Reply:
"we can be more positive" is not "we are certain". The difference between these two statements is what I mean by "sweeping the problem under the rug". You don't solve the problem, you just make it less likely to show up.

Reply:
@alonSadan if you look at https://github.com/kubevirt/kubevirt/pull/4010/commits/cc62349d370c39e4fc5776a12cbd3e1436b4c32b#diff-777ec290af00beeee5968636afc39636fc0399afd3818df5f6efba35bce80419R425 you will see that with the timeout increase the test does actually not test anymore what it is supposed to test. The test waits for 10 seconds with the assumption that the 1m process will have a regular runtime of less than 10 seconds.

@dankenigsberg I agree that this needs another mechanism to test things. However we tend to test, if possible, always end-to-end, so only asking libvirt for the status is not something I would call optimal. Taking uptime or something, like you suggested, sounds like a good plan.

Reply:
> @alonSadan if you look at cc62349#diff-777ec290af00beeee5968636afc39636fc0399afd3818df5f6efba35bce80419R425 you will see that with the timeout increase the test does actually not test anymore what it is supposed to test. The test waits for 10 seconds with the assumption that the 1m process will have a regular runtime of less than 10 seconds.


@rmohr  yes, I totally missed those two waiting lines. I suggested a quick-fix #4394.

> we can be more positive" is not "we are certain". The difference between these two statements is what I mean by "sweeping the problem under the rug". You don't solve the problem, you just make it less likely to show up

@dankenigsberg  I understand. My reason was to apply the safe expecter with changing as less as possible the existing tests. so I tried to keep the test mechanism, but make it pass with the safe expecter. This test didn't worked , and the old expecter coverd for it. (after  #4394. will get in) The test will still not be perfect, but at least better then it was.
to improve the tests mechanism I would suggest a different PR.

Reply:
@dankenigsberg @rmohr 
Eventually I removed the old test and added a new one based on the uptime diff of the guest and the host.
you can see it here  #4394.
Thanks for the help.